timeseries/src/autogluon/timeseries/utils/features.py

```diff
@@ -73,7 +74,7 @@ class ContinuousAndCategoricalFeatureGenerator(PipelineFeatureGenerator):
     Imputes missing categorical features with the most frequent value in the training set.
     """
 
-    def __init__(self, verbosity: int = 0, minimum_cat_count=2, float_dtype: str = "float64", **kwargs):
```

Comment:
Why not keep the `float_dtype` param and default it ti float32 instead?

Reply:
I've moved the logic for converting numeric columns to `float_dtype` from `ContinuousAndCategoricalFeatureGenerator` (a separate instance is created for past, known, and static covariates) to `TimeSeriesFeatureGenerator` (a single object processes all columns). This way we reduced the number of `df.astype()` calls from 2 to 1, decreasing the runtime and reducing the memory usage.